libavfilter/vf_signature.c

```diff
@@ -748,6 +758,310 @@ static int config_output(AVFilterLink *outlink)
     return 0;
 }
 
+
+static void release_streamContext(StreamContext *sc)
+{
+	free(sc->coarsesiglist);
+	free(sc->finesiglist);
+}
+static int finesignaturecompare(const void* p1, const void* p2)
+{
+	FineSignature *a = (FineSignature*)p1;
+	FineSignature *b = (FineSignature*)p2;
+	if (a->pts == b->pts)
+		return 0;
+	else if (a->pts < b->pts)
+		return -1;
+	else
+		return 1;
+}
+static int get_filesize(const char *filename)
+{
+	int fileLength = 0;
+	FILE *f = NULL;
+	f = fopen(filename, "rb");
+	fseek(f, 0, SEEK_END);
+	fileLength = ftell(f);
+	fclose(f);
+	return fileLength;
+}
+static uint8_t * get_filebuffer(const char *filename, int* fileLength)
+{
+	FILE *f = NULL;
+	unsigned int rResult = 0, paddedLength = 0;
+	uint8_t *buffer = NULL;
+	GetBitContext bitContext = { 0 };
+	//check input parameters
+	if (strlen(filename) <= 0) return buffer;
+	f = fopen(filename, "rb");
+	if (f == NULL) return buffer;
+
+	*fileLength = get_filesize(filename);
+
+	// Cast to float is necessary to avoid int division
+	paddedLength = ceil(*fileLength / (float)AV_INPUT_BUFFER_PADDING_SIZE)*AV_INPUT_BUFFER_PADDING_SIZE + AV_INPUT_BUFFER_PADDING_SIZE;
+	buffer = (uint8_t*)calloc(paddedLength, sizeof(uint8_t));
+
+	// Read entire file into memory
+	rResult = fread(buffer, sizeof(uint8_t), *fileLength, f);
+	fclose(f);
+	f = NULL;
+    return buffer;
+}
+
+static int binary_import(uint8_t *buffer, int fileLength, StreamContext *sc)
+{
+	int ret = 0;
+
+	unsigned int numOfSegments = 0;
+	GetBitContext bitContext = { 0 };
+
+	// BE CAREFUL, THE LENGTH IS SPECIFIED IN BITS NOT BYTES
+	init_get_bits(&bitContext, buffer, 8 * fileLength);
+	// libavcodec
+
+	// Skip the following data:
+	// - NumOfSpatial Regions: (32 bits) only 1 supported
+	// - SpatialLocationFlag: (1 bit) always the whole image
+	// - PixelX_1: (16 bits) always 0
+	// - PixelY_1: (16 bits) always 0
+	skip_bits(&bitContext, 32 + 1 + 16 * 2);
+
+
+	// width - 1, and height - 1
+	// PixelX_2: (16 bits) is width - 1
+	// PixelY_2: (16 bits) is height - 1
+	sc->w = get_bits(&bitContext, 16);
+	sc->h = get_bits(&bitContext, 16);
+	++sc->w;
+	++sc->h;
+
+	// StartFrameOfSpatialRegion, always 0
+	skip_bits(&bitContext, 32);
+
+	// NumOfFrames
+	// it's the number of fine signatures
+	sc->lastindex = get_bits_long(&bitContext, 32);
+
+	// sc->time_base.den / sc->time_base.num
+	// hoping num is 1, other values are vague
```

Comment:
If this works then fine, if not we can try using input file timebase here as that is what the filter used to generate these values. Unclear to me which timebase it prefers out of the two inputs.
https://github.com/livepeer/FFmpeg/blob/b344789a31a2d806be702239d7614f7a9f51f941/libavfilter/vf_signature.c#L90-L96

Reply:
It works fine now.